---
title: "Construct SCE for data in brief"
author: "Daniel Brown"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float:
      smooth_scroll: true
    theme: readable
    highlight: tango 
    df_print: paged
    code_folding: hide
editor_options: 
    chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(scater)
library(scuttle)
library(here)
library(tidyverse)
library(DropletUtils)
```

# Construct single cell experiment object for data in brief

## Aim

After running cellranger 7.0.1 multi on each of the 4 batches I subsequenctly ran cellranger agrr to combine all the samples into a single object.

Add the various other cell metadata in this notebook for submission to data in brief.

## 10x Genomics Flex data

```{r read_sce}
dir <- "/vast/scratch/users/brown.d/data_in_brief/WB_CSL/outs/count/filtered_feature_bc_matrix/"

sce <- DropletUtils::read10xCounts(samples = dir)
colnames(sce) <- sce$Barcode

sce
```

## Save the basic SCE

```{r save_base, eval=FALSE}
sce_orig <- sce
saveRDS(sce,
        "/stornext/Projects/GenomicsRnD/brown.d/G000254_fixed_neutrophil/SCEs/flex_basic.sce")
```

## Add metadata details

Add in the metrics about fixation time, number of cells fixed etc.

```{r, eval=FALSE}
aggregaation <- read.delim("/stornext/Projects/GenomicsRnD/brown.d/G000254_fixed_neutrophil/WB_CSL/outs/aggregation.csv", sep=",")
metadata <- as.data.frame(str_split(aggregaation$sample_id, pattern = "_", simplify = T))
colnames(metadata) <- c("Batch", "ID", "Fix_num", "Probe_num")

# Write to file and I will manually fix up
write.csv(metadata, "/stornext/Projects/GenomicsRnD/brown.d/G000254_fixed_neutrophil/WB_CSL/outs/metaadata.csv")
```

Read in my manually munged cell metadata

```{r}
metadata <- read_csv("/stornext/Projects/GenomicsRnD/brown.d/G000254_fixed_neutrophil/WB_CSL/outs/metaadata.csv")
metadata$Capture <- as.factor(metadata$Capture)
tb <- as_tibble(colData(sce))
tb$ID <- colnames(sce)
tb$Capture <- as.factor(str_split(string = tb$ID, pattern = "-", simplify = TRUE)[,2])
```

Join the metadata to the coldata

```{r}
mdt <- right_join(metadata, tb) %>% 
  select(-ID) %>% 
  relocate(Barcode) %>% 
  relocate(Sample)
```

Add back the metadata as colData to the single cell experiment object

```{r}
colData(sce) <- DataFrame(mdt)
```

## Change row names back to gene symbols

```{r}
rownames(sce) <- rowData(sce)$Symbol
```

## Compute QC metrics

Add per cell QC adds the QC info to the colData of the sce.

```{r}
is.mito <- grepl("^MT", rownames(sce))
sce <- addPerCellQC(sce, subsets=list(
  Mito=is.mito))
```

Add experimental metdata

```{r}
metadata(sce)$Platform <- "10x Genomics Flex v1"
```


### Save SCE with quality control

```{r}
saveRDS(sce,
        "/stornext/Projects/GenomicsRnD/brown.d/G000254_fixed_neutrophil/SCEs/flex_qcMetrics.sce")
```

## Session Info

```{r}
sessionInfo()
```
