---
title: "Examine metadata for data in brief"
author: "Daniel Brown"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float:
      smooth_scroll: true
    theme: readable
    highlight: tango 
    df_print: paged
    code_folding: hide
editor_options: 
    chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(scuttle)
library(knitr)
library(tidyverse)
```

# Examine metadata for data in brief

## Aim

To generate some of the figures and tables for data in brief, make some summary metrics on number of batches, cells etc.

## Read data

```{r read_sces}
hive <- readRDS(here::here(
  "data/SCEs/hive_qcMetrics_sce.rds"
  ))
  
flex <- readRDS(here::here(
  "data/SCEs/flex_qcMetrics.sce.rds"
  ))
```

Combine the coldata together

```{r}
keep_cols <- intersect(
  colnames(colData(hive)),
  colnames(colData(flex))
)

tb <- as_tibble(rbind(
  colData(hive)[,keep_cols],
  colData(flex)[,keep_cols]
))

# The Flex metadata has more detail as there are many more batches and cells
flex_cd <- as_tibble(colData(flex))
```

### Metadata description

* Sample
  + The path to 10x Genomics cellranger output on my machine
* Capture
  + The ID of the capture (physical 10x Genomics microfluidics chip).
  + For 10x Genomics Flex batch 3, we received a replacement set of library preparation reagents due to poor performance of this batch (“batch 3A”).
  + As instructed by a 10x Genomics field application specialist we performed a second capture reaction of remaining material from batch 3A = “batch 3B
* Fix_num
  + The number of cells fixed
* Probe_num
  * The number of cells used to hybridise gene specific probes in 10x Genomics Flex experiments
*  Storage_time_weeks
  + The number of weeks from the fixation of the samples until library preparation and scRNA-Seq.
* RBC_debris
  + Whether red blood cell debris was visible in the fixed samples. Our hypothesis is that RBC debris negatively affects data quality
* Donor
  + The de-identified code for the healthy blood donor
  
The following metadata fields were generated by scuttle::addPerCellQC()
  
* sum
  + The total number of unique molecular identifiers (UMIs) detected in the cell. Also known as library size.
* detected
  + The number of genes detected in the cell
* subsets_Mito_sum
  + The total number of unique molecular identifiers (UMIs) from mitochondrial genes (^MT)
* subsets_Mito_detected
  + The number of mitochondrial genes detected
* subsets_Mito_percent
  + The percent of mitochondrial UMIs of the total library size. Often used as a quality control metric for low quality cells
* total
  + Same meaning as sum field

## Summarise metadata

How many cells are in the dataset? Specifically cell barcodes, cell calling has not been comprehensively performed.

```{r}
n_cells <- nrow(tb)
n_samples <- tb %>% 
  dplyr::count(Donor, sort=TRUE)
```

For both HIVE and Flex datasets there are:

* Cells: `r n_cells`
* Samples : `r nrow(n_samples)`

```{r}
n_batch <- flex_cd %>% 
  count(Batch)
n_capture <- flex_cd %>% 
  count(Capture)
```

For Flex there are:

* Batches : `r nrow(n_batch)`
* Samples `r nrow(n_capture)`

Make a nice table summarizing the key metadta for Flex experiments

```{r}
flex_cd %>% 
  count(Batch, Donor) %>% 
  pivot_wider(names_from = Batch, values_from = n) %>% 
  kable()
```

## Session Info

```{r}
sessionInfo()
```
